core/modules/ccontrol/UserQueryResources.h

```diff
@@ -81,7 +88,7 @@ class UserQuerySharedResources {
 
     UserQuerySharedResources(UserQuerySharedResources const& rhs) = default;
     UserQuerySharedResources& operator=(UserQuerySharedResources const& rhs) = delete;
-
+    czar::CzarConfig const& czarConfig;
```

Comment:
czarConfig is a member of Czar and has the same lifetime as the czar itself. If it has gone missing, something is seriously wrong. A shared pointer cannot be used to pass it since Czar::czarConfig wasn't created that way.